test/backend/api/v1/ChatTest.spec.ts

```diff
@@ -1,25 +1,86 @@
-import chai from 'chai';
+import chai, {expect} from 'chai';
+import {Chat} from "../../../../src/backend/Models";
+import mongoose = require("mongoose");
 import chaiHttp = require('chai-http');
-import {app} from "../../../../src/App";
-import {expect} from "chai";
+
 chai.use(chaiHttp);
 
 
 describe('Chats', () => {
+    let app: any;
+    before(async() => {
+        process.env.DB_CONNECTION_STRING = 'mongodb://localhost:27017/test_project';
+        app = require('../../../../src/App').app;
+        await mongoose.connect(process.env.DB_CONNECTION_STRING as string, {useNewUrlParser: true});
+        mongoose.connection.on('error', () => expect.fail('Error connecting to db'));
+    });
 
     describe('Get all chats', () => {
-        // TODO: Add test with mongodb
-        it('it should GET all the chats', (done) => {
-            chai.request(app)
+
+
+        before(async ()=>{
+            const clear_chat = Chat.deleteMany({});
+            await clear_chat.exec();
+            await Chat.create({senderUsername: 'test1',
+                receiverUsername: 'test2',
+                content: 'This is a test message',
+                time: new Date()});
+        });
+
+        it('it should GET all the chats', async () => {
+            return chai.request(app)
                 .get('/api/v1/chats')
+                .query({sender_id: 'test1', receiver_id: 'test2'})
                 .then((res) => {
                     expect(res).have.status(200);
                     expect(res.body).have.key('allMessages');
                     expect(res.body.allMessages.length).eql(1);
-                    done();
                 }).catch((err) => {
-                    throw err;
-            });
+                    expect.fail('should not throw error' + err);
+            })
+        });
+
+        it('it should raise bad request if no query specified', async () => {
+            return chai.request(app)
+                .get('/api/v1/chats')
+                .query({aaa: 'test1', bbb: 'test2'})
+                .then((res) => {
+                    expect(res).have.status(400);
+                    expect(res.body).have.key('message');
+                    expect(res.body.message).eql('sender_id or receiver_id not found');
+                }).catch((err) => {
+                    expect.fail('should not throw error' + err);
+                })
+        });
+    });
+
+
+    describe('Get all chats2', () => {
```

Comment:
Why "Get all chats2"? Shouldn't it be something like "Post a chat"?

Reply:
I forgot to change this when I tried to debug the app & mongoose connection between tests. 😢